test/new-e2e/examples/customenv_with_two_vm_test.go

```diff
@@ -6,49 +6,195 @@
+	"github.com/DataDog/test-infra-definitions/components/docker"
 	"github.com/DataDog/test-infra-definitions/resources/aws"
 	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
 
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
+//go:embed "testfixtures/http_compose.yaml"
+var httpbinComposeContent string
+
 type multiVMEnv struct {
-	MainVM *components.RemoteHost
-	AppVM  *components.RemoteHost
+	// This is initialised by the run function
+	MainVM     *components.RemoteHost
+	AppVM      *components.RemoteHost
+	FakeIntake *components.FakeIntake
+	// This is initialised by the Init function
+	AppDocker *client.Docker
+	Agent     *agentclient.Agent
+}
+
+// version one with explicit exports
+func multiVMEnvProvisionerOne() e2e.PulumiEnvRunFunc[multiVMEnv] {
+	return func(ctx *pulumi.Context, env *multiVMEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+
+		// We create a first VM where we install a workload application using docker
+		// We use os.AmazonLinuxECSDefault to have linux installed with docker
+		appHost, err := ec2.NewVM(awsEnv, "app", ec2.WithOS(compos.AmazonLinuxECSDefault))
+		if err != nil {
+			return err
+		}
+		// we export the app vm to the suite's environment
+		// this will export the remote host IP address, the username
+		// the OS and other useful information
+		appHost.Export(ctx, &env.AppVM.HostOutput)
+		// initialize a docker manager without installing docker.io (that false)
+		dockerManager, _, err := docker.NewManager(*awsEnv.CommonEnvironment, appHost, false)
+		if err != nil {
+			return err
+		}
+		composeContents := []docker.ComposeInlineManifest{docker.ComposeInlineManifest{
+			Name:    "httpbin",
+			Content: pulumi.String(httpbinComposeContent),
+		}}
+		_, err = dockerManager.ComposeStrUp("httpbin", composeContents, pulumi.StringMap{})
+		if err != nil {
+			return err
+		}
+
+		// we deploy a fakeintake
+		fakeIntake, err := fakeintake.NewECSFargateInstance(awsEnv, "fakeintake")
+		if err != nil {
+			return err
+		}
+		err = fakeIntake.Export(ctx, &env.FakeIntake.FakeintakeOutput)
+
+		// Finally we create a second VM where we install the agent
+		mainHost, err := ec2.NewVM(awsEnv, "main")
+		if err != nil {
+			return err
+		}
+		mainHost.Export(ctx, &env.MainVM.HostOutput)
+		// we install the agent passing the reference to the fakeintake
+		_, err = agent.NewHostAgent(awsEnv.CommonEnvironment, mainHost, agentparams.WithFakeintake(fakeintake))
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
 }
 
-func multiVMEnvProvisioner() e2e.PulumiEnvRunFunc[multiVMEnv] {
+// version two with self-exporting components
+func multiVMEnvProvisionerTwo() e2e.PulumiEnvRunFunc[multiVMEnv] {
 	return func(ctx *pulumi.Context, env *multiVMEnv) error {
+		// first, let's create an aws environment
+		// this is where our remote hosts will live
 		awsEnv, err := aws.NewEnvironment(ctx)
```

Comment:
We probably want to reuse the same `awsEnv` than in `multiVMEnvProvisionerOne()`